app/connectors/GoodsRecordConnector.scala

```diff
@@ -67,12 +68,38 @@ class GoodsRecordConnector @Inject() (config: Configuration, httpClient: HttpCli
 
   private def searchRecordsUrl(
     eori: String,
-    searchTerm: String,
+    searchTerm: Option[String],
     exactMatch: Boolean,
     queryParams: Map[String, String]
   ) =
     url"$dataStoreBaseUrl/trader-goods-profiles-data-store/traders/$eori/records/filter?searchTerm=$searchTerm&exactMatch=$exactMatch&$queryParams"
 
+  private def filterSearchRecordsUrl(
+    searchTerm: Option[String],
+    exactMatch: Boolean,
+    countryOfOrigin: Option[String],
+    immiReady: Option[Boolean],
+    notReadyForIMMI: Option[Boolean],
+    actionNeeded: Option[Boolean],
+    queryParams: Map[String, String]
+  ): URL = {
+
+    val queryParamsSeq = Seq(
+      searchTerm.map(term => s"searchTerm=$term"),
+      Some(s"exactMatch=$exactMatch"),
+      countryOfOrigin.filter(_.nonEmpty).map(origin => s"countryOfOrigin=$origin"),
+      immiReady.map(ready => s"immiReady=$ready"),
```

Comment:
Had a look in the UI and other other filters work except the immiReady one I think the casing is slightly different, I believe it should be

IMMIReady